x/wasm/keeper/keeper.go

```diff
@@ -187,16 +187,20 @@ func (k Keeper) create(ctx sdk.Context, creator sdk.AccAddress, wasmCode []byte,
 		return 0, checksum, sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "cannot be nil")
 	}
 
-	if !authZ.CanCreateCode(k.getUploadAccessConfig(ctx), creator) {
-		return 0, checksum, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "can not create code")
-	}
-	// figure out proper instantiate access
 	defaultAccessConfig := k.getInstantiateAccessConfig(ctx).With(creator)
 	if instantiateAccess == nil {
 		instantiateAccess = &defaultAccessConfig
-	} else if !instantiateAccess.IsSubset(defaultAccessConfig) {
-		// we enforce this must be subset of default upload access
-		return 0, checksum, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "instantiate access must be subset of default upload access")
+	}
+
+	if !authZ.IsGovAuthorization() {
```

Comment:
This is the switch that I wanted to avoid by moving the conditions below fully into the  policies. My motivation was the [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns). The AuthorizationPolicies are an extension point where custom implementations can be used for individual decision logic. This would not be possible if we keep the code here.
Ideally we have a 3 liner here like:
```
if !authZ.CanCreateCode(chainDefaults, creator, instantiateAccess) {
	return 0, checksum, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "can not create code")
}
```
I hope this makes sense for you, too

Reply:
@alpe thanks for your suggestion. Is it fine if I move the subset check into CanCreateCode like this ?: 
```
func (p DefaultAuthorizationPolicy) CanCreateCode(config types.AccessConfig, actor sdk.AccAddress, defaultConfig types.AccessConfig) bool {
	isSubset := true
	if !config.IsSubset(defaultConfig) {
		isSubset = false
	}
	return config.Allowed(actor) && isSubset
}
```

Reply:
I've force-pushed my code into this branch to avoid having too much PR!

Reply:
tysm for your suggestion!